### ISSUE TITLE
[BUG] Vislib to Vega-lite migrator, fix chart overflow issue

### ISSUE THREAD

**Issue by lezzago** [opensearch-project/OpenSearch-Dashboards#3485]
**Describe the bug**

For charts that are being rendered with vega-lite instead of vislib, the chart has an overflow error where you need to scroll in the chart to view the entire chart as there is some css styling problem that doesnt size the chart correctly when displaying on a visualize embeddable. This is related to the #2880 feature and came from the #3106 PR.

**To Reproduce**
Create a line chart that is rendered by vega-lite

**Expected behavior**
Have the chart fix in the embeddable object without the need to scroll.

**Comment by ohltyler** [comment]
It looks that autosize isn't working correctly. When calling `autosize()` on the `view` itself, we can update to have it show correctly.

For example, adding below line in `updateVegaSize()` will resolve the problem.
```
       view.autosize({
          type: 'fit',
          contains: 'padding',
        })
```

Still needs to be researched more since there's no documentation on this

**Comment by ohltyler** [comment]
Adding things here as I find them:
- the vega lite spec (set to `vlspec` in vegaparser) is compiled in vega_fn (the vega expr fn)
- even with a set `autosize` value in the vega spec, in the context of `vconcat`, this is getting ignored. For example, setting the spec `autosize` field to `{ type: 'fit', contains: 'padding'}` is ignored, and the default value of `{type: 'pad', contains: 'padding'}` is set in the vega spec (the compiled spec generated from the _vega lite_ spec that is controlled by us
- there is a `autosize` signal provided in the vega view (unable to find docs on this method being used or existing). when updating that with our desired autosize values of `{type: 'fit', contains: 'padding'}`, this updates our view correctly, and fixes the scrolling issues, and resizing the window etc. resizes the chart as expected with no overflow

**Comment by ohltyler** [comment]
To summarize, given the [autosize documentation](https://vega.github.io/vega/docs/specification/) for `vega` (_not_ `vega-lite`), there is no mention of limitations on multi-view/faceted displays not working, etc. That is because there is no notion of `vconcat/`hconcat`/ etc. at the vega level.

So, if we set this setting after compilation in the vega spec, it allows autosize to work correctly for our multi-view `vconcat` _vega-lite_ spec.

**Comment by ohltyler** [comment]
This will be resolved as part of #3415

**Comment by ohltyler** [comment]
Closed as part of #3415
